VOSA-496 OpenJDK support hardening

skybert opened the pull request:
- Put APT credentials in /etc/apt/auth.conf instead of sources.list
- environment.java_home now skips Java installation for both OpenJDK
  and Oracle JDK (before it only skipped it for Oracle JDK).
- '/usr/bin/ece start' now works on OpenJDK 9 too
- clean-slate integration test does incremental sleeps while waiting
  for the vbox machine to come up
- ece-install support for OpenJDK 9
- fixed OpenJDK 10 unit test
- documented the most common ece-install.yaml settings in
  ece-install-guide.org
- fallback JDK package on Debian based machines is now
  default-jdk-headless
- fixed java::_java_get_spec_version, can't run java process in run
  wrapper as it will not return any value.

Commit:
VOSA-496 OpenJDK support hardening

- Put APT credentials in /etc/apt/auth.conf instead of sources.list
- environment.java_home now skips Java installation for both OpenJDK
  and Oracle JDK (before it only skipped it for Oracle JDK).
- '/usr/bin/ece start' now works on OpenJDK 9 too
- clean-slate integration test does incremental sleeps while waiting
  for the vbox machine to come up
- ece-install support for OpenJDK 9
- fixed OpenJDK 10 unit test
- documented the most common ece-install.yaml settings in
  ece-install-guide.org
- fallback JDK package on Debian based machines is now
  default-jdk-headless
- fixed java::_java_get_spec_version, can't run java process in run
  wrapper as it will not return any value.

mogsie commented on usr/bin/ece:
```diff
@@ -351,8 +351,10 @@ function set_instance_settings() {
         ece_args=$ece_args" "$gc_rolling_args
       fi
     elif is_java_vendor_openjdk; then
+      # Works on both OpenJDK 8 and 11, if we no longer need OpenJDK
```
It also works on Oracle JDK 8 / 11?

skybert replied:
Yes. We add it for `is_java_vendor_oracle` too, a bit further up in `/usr/bin/ece`.

mogsie commented on usr/sbin/ece-install:
```diff
@@ -324,10 +324,12 @@ function install_common_os_packages()
       local package_pool=${fai_apt_vizrt_pool-stable}
-      add_apt_source "deb https://${fai_package_apt_user}:${fai_package_apt_password}@apt.escenic.com ${package_pool} main non-free"
+      add_apt_source "deb https://apt.escenic.com ${package_pool} main non-free"
+      echo machine apt.escenic.com \
```
Just to be 100% sure perhaps we should append a line break to this file before appending our line?  Just in case someone made a file without a trailing newline.

`(echo; echo machine apt.escenic.com \ ...... ) >> /etc/apt/auth.conf`

Commit:
VOSA-496 Hardening: Wait for APT locks

Wait for a certain amount of time to ensure there's no other APT
process running. systemd may be running an update process on boot of
new machines or there may be a daemon like unattended-upgrades running
which also grabs the APT lock file(s).

Commit:
Better default presentation indexer-webservice

- if installing search and editor on the same machine, the
  presentation indexer webservice URI required the user to specify:

  profiles:
    presentation:
      search_indexer_ws_uri: http://localhost:8080/indexer-webservice/presentation-index/

  otherwise
  /opt/tomcat-search1/conf/Catalina/localhost/indexer-webapp-presentation.xml
  got the port number of the appserver currently installing, which was
  search's port, which is by default 8180. This should rather be the
  port number of the editor instance.

- the configuration option profiles.presentation.search_indexer_ws_uri
  still works.
